### ISSUE TITLE
Exported model translation is different from the original checkpoint

### ISSUE THREAD

**Issue by anasir-ureed** [OpenNMT/OpenNMT-tf#501]
Greetings,

I am trying to evaluate a model I have been training. When I use onmt-main infer, I got some results, which I am somehow satisfied with. I try to export the model, and then use it with tensorflow serving. The results are somehow different. I tried some segments using the exported version and the checkpoint version using infer command, the result translations for the same source sentence are different when I use the exported version compared to the checkpoint version. 

They both are coming from the same checkpoint! Is that expected? What do you think?

Please advise,
Best wishes,

**Comment by guillaumekln** [comment]
Hi,

Outputs should be the same. What are the differences you observe?

Also, how did you export the model and run the inference to be sure that the same checkpoint is used in both cases?

**Comment by RubaJaikat** [comment]
Hello, 

what we did is we have trained a model and evaluated it against a specific testset. 
`python -m opennmt.bin.main infer --config config.yaml --features_file /path/to/testset --predictions_file /path/to/prediction_file --checkpoint_path /model/model.ckpt-214896`

So, we get translations for the testset saved in "prediction_file"

Then we export that same checkpoint to use in tensorflow-serving as follows: 
`python -m opennmt.bin.main export --export_dir_base /path/to/export/dir --auto_config --config config.yaml --checkpoint_path /model/model.ckpt-214896`

Now, when sending requests to the docker image which we included the exported model in to translate the same testset, we get different translations. 
What could caused the issue?

**Comment by guillaumekln** [comment]
You used `--auto_config` during export, but not inference. This could explain the difference.

**Comment by RubaJaikat** [comment]
It solved the issue! Thank you !

**Comment by anasir-ureed** [comment]
@guillaumekln 
Sorry, false alarm!

We actually tested the checkpoint 4 different ways, each one led to different results.
1. infer with auto_config
2. infer without auto_config
3. export with auto_config
4. export with auto_config

When evaluating the whole test set, none of the above options matched with any other one.

**Comment by guillaumekln** [comment]
How much differences are we talking about?

Also make sure that the TensorFlow Serving version matches the TensorFlow version and that you are using the same device type for both runs (e.g. run both on GPU or both on CPU).

**Comment by anasir-ureed** [comment]
Sorry, another false alarm.
The issue is resolved totally. 

In the reopen comment we had an issue in the inference script we are using. We were calling another API by mistake.

Thanks for your help and assistance.

Best wishes :)
